cekit/generator/osbs.py

```diff
@@ -65,26 +67,48 @@ def prepare_artifacts(self):
         and fetches all of them
         """
 
-        logger.info("Handling artifacts...")
+        logger.info("Handling artifacts for OSBS...")
         target_dir = os.path.join(self.target, 'image')
         fetch_artifacts_url = []
 
         for image in self.images:
             for artifact in image.all_artifacts:
-                logger.info("Preparing artifact '{}'".format(artifact['name']))
-
-                if isinstance(artifact, _PlainResource) and \
-                        config.get('common', 'redhat'):
+                logger.info("Preparing artifact '{}' (of type {})".format(artifact['name'], type(artifact)))
+
+                if isinstance(artifact, _UrlResource):
+                    intersected_hash = [x for x in crypto.SUPPORTED_HASH_ALGORITHMS if x in artifact]
+                    logger.debug("Found checksum markers of {}".format(intersected_hash))
+                    if not intersected_hash:
+                        logger.warning("No md5 supplied for {}, calculating from the remote artifact".format(artifact['url']))
+                        intersected_hash = ["md5"]
+                        tmpfile = tempfile.NamedTemporaryFile()
+                        try:
+                            artifact.download_file(artifact['url'], tmpfile.name)
+                            artifact["md5"] = crypto.get_sum(tmpfile.name, "md5")
+                        finally:
+                            tmpfile.close()
+
+                    fetch_artifacts_url.append({'url': artifact['url'],
```

Comment:
You don't add the computed md5 here?

Reply:
Ah, it's added later: https://github.com/cekit/cekit/pull/661/files#diff-53fb5968a14cd97636475ff271beedac8d436a8e4a42bbd44235de8e37307977R93